openedx/core/djangoapps/timed_notification/tasks.py

```diff
@@ -0,0 +1,63 @@
+import logging
+
+from datetime import datetime, timedelta, date
+from pytz import utc
+
+from core import send_course_notification_email, get_course_link
+from lms.djangoapps.branding import get_visible_courses
+
+from celery.task import periodic_task
+from celery.schedules import crontab
+
+from django.db import close_old_connections
+
+log = logging.getLogger('edx.celery.task')
+
+
+@periodic_task(run_every=crontab(minute=1, hour=0))
+def execute_dialy():
+    log.info("TIMED-NOTIFICATION-JOB execute_dialy")
+    close_old_connections()
+    courses = get_visible_courses()
+
+    while courses:
+        course = courses[-1]
+
+        # we're interested only in dates
+        course_start_date = course.start.date()
+        date_now = datetime.now(utc).date()
+
+        # Email-template context
+        context = {
+            'course_name': course.display_name,
+            'course_link': get_course_link(course_id=course.id),
+        }
+
+        # check if course is already started or just started today
+        if date_now >= course_start_date:
```

Comment:
Could you please explain when you want to send this notification.  right now it appears you want to send notification on every monday after course is started and it will continue until doomsday.

Reply:
Ops, my bad. I must  extend this check with <code>and date_now <= course_end_date</code> .